scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -43,6 +43,24 @@ class FormatWriter(formatOps: FormatOps) {
                 .getOrElse(token.syntax, token.syntax)
             sb.append(rewrittenToken)
         }
+
+        if (runner.dialect.allowTrailingCommas &&
+          state.splits.last.modification.isNewline &&
```

Comment:
It would be good to have a parallel that removes the trailing comma if `!modification.isNewline`

Reply:
Yep, it may be necessary for handling `configStyleArguments = false`

Reply:
also necessary for
```scala
foo(a,
  b,
)
```

Reply:
yep, I have a skipped test for it. Thanks for the examples, it's hard to come up with those :D

Reply:
The comma should btw be removed in that case regardless of the setting since it's a bug to produce `foo(a, b,)`